advanced-creation.md

```diff
@@ -114,3 +114,10 @@ const unicorn = got.create(settings);
 // Same as:
 const unicorn = got.extend({headers: {unicorn: 'rainbow'}});
 ```
+
+##### preventChanges
```

Comment:
I'm not sure we want to allow things to be mutated by default, or via a configuration for that matter. Why was this necessary or even wanted?